server/covmanager/cron.py

```diff
@@ -0,0 +1,76 @@
+import logging
+import requests
+
+from celeryconf import app
+from dateutil.relativedelta import relativedelta, MO
+from django.conf import settings
+from django.db.models import Q
+
+from crashmanager.models import Client
+from .models import Collection, Repository
+from .tasks import aggregate_coverage_data
+
+logger = logging.getLogger("covmanager")
+
+
+# The `create_current_weekly_report_mc` task and dependencies are part of the
+# internal process at Mozilla to create an aggregated fuzzing coverage report
+# on a weekly basis. For this purpose, a revision is fixed every week and all
+# fuzzing tools perform a coverage run on the that revision (during the week).
+# At the end of the week, the code below fetches all reports belonging to that
+# revision (which is published as an URL) and aggregates them into a combined
+# weekly report. You can use the same logic if you have a large project with
+# multiple testing tools and would like to automate the process of generating
+# a summarized report for your testing efforts.
+
+
+def create_weekly_report_mc(revision):
+    # Some of our builds (e.g. JS shell) use the HG short revision format
+    # to submit their coverage while the server provides the full revision.
+    short_revision = revision[:12]
+
+    repository = Repository.objects.get(name="mozilla-central")
+    client = Client.objects.get_or_create(name="Server")
+
+    collections = Collection.objects.filter(
+        Q(revision=revision) | Q(revision=short_revision)).filter(repository=repository)
+
+    last_monday = collections.first().created + relativedelta(weekday=MO(-1))
```

Comment:
If you want the earliest collection, you should add an explicit `order_by()` clause so we aren't relying on a DB dependent ordering.